### ISSUE TITLE
workspace dependency breaks default-features = false

### ISSUE THREAD

**Issue by Follpvosten** [async-graphql/async-graphql#1236]
## Expected Behavior
`default-features = false` on async-graphql should turn off the email-validator, tempfile, playground and graphiql features on async-graphql.

## Actual Behavior
it does – as long as you don't depend on any other projects from the async-graphql workspace.

## Steps to Reproduce the Problem
1. create an empty cargo project and add async-graphql with default-features = false to the dependencies.
2. also add a dependency on, for example, async-graphql-axum (which has default-features = false on the workspace dependency, but this does not appear to work)
3. cargo check once, then check the Cargo.lock file to find unexpected packages in there, like tempfile or handlebars.

## Specifications

- Version: 5.0.6 – confirmed against git master as well
- Platform: any
- Subsystem: any

## Possible solutions
I have verified that locally, I can fix this by changing one line in the workspace Cargo.toml, namely

```diff
[workspace.dependencies]
-async-graphql = { path = ".", version = "5.0.6" }
+async-graphql = { path = ".", version = "5.0.6", default-features = false }
```

however, this is obviously a major breaking change because it will mean that `default-features = false` works again, so I'm not sure how to proceed here. I also don't know if this change will break anything else.

**Comment by frederikhors** [comment]
I think this can be closed?

**Comment by Follpvosten** [comment]
oh great it *was* released as a minor point release despite being a major breaking change :sweat:

**Comment by Follpvosten** [comment]
to make more clear what I mean: there is a *public struct field* in UploadValue (`content`) which *changes type* based on a feature: https://github.com/async-graphql/async-graphql/blob/master/src/types/upload.rs#L11

someone accessing that field directly may have their code randomly break if they have `default-features = false` suddenly work correctly again

**Comment by Follpvosten** [comment]
this was also *not the case in 4.0* (as the `tempfile` feature didn't exist yet)

**Comment by Follpvosten** [comment]
I suggest the latest release be yoinked immediately as to not break people's code when they run `cargo update` @sunli829

**Comment by frederikhors** [comment]
I think adding a feature in Cargo.toml is not a big issue. Even if this can be called breaking change. Is not something to fix in code (which is definitely more challenging). Am I wrong?

**Comment by Follpvosten** [comment]
Yeah if you explicitly *don't* care about following semver, putting this out in a minor release is fine of course.

I'm gonna try to explain better with an example.

Imagine a project with this in the dependencies:

```toml
async-graphql = { version = "5.0", default-features = false }
```

with a graphql handler like this:

```rust
async fn create_asset(&self, ctx: &Context<'_> file: Upload) -> Result<Uuid> {
    let file = file.value(ctx)?.content;
    let content = tokio::task::spawn_blocking(move || -> Result<_> {
        let mut buf = Vec::with_capacity(size);
        file.read_to_end(&mut buf)?;
        Ok(buf)
    })
    .await??;
    do_something_else(&content).await
}
```

when 5.0.6 is in the lockfile, this will compile fine (`content` is of type `File`).

once the user runs `cargo update`, which is not expected to break your code, the lockfile will depend on 5.0.7 it will no longer compile (because `content` is of type `Bytes` now).

**Comment by Follpvosten** [comment]
closing in favour of #1263 I guess
